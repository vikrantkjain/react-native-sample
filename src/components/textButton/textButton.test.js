import {fireEvent, render, screen} from '@testing-library/react-native';
import TextButton from './index';
import renderer from 'react-test-renderer';
const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};
describe('Test TextButton', () => {
  const onPressMock = jest.fn();
  const tree = renderer.create(
    <TextButton title="hey" onPress={onPressMock} />,
  );
  render(<TextButton title="hey" onPress={onPressMock} />);
  it('Render successfully', () => {
    expect(tree).toBeDefined();
  });
  it('Render correctly', () => {
    const treeToJSON = tree.toJSON();
    expect(treeToJSON).toMatchSnapshot();
  });
  fireEvent.press(screen.getByTestId('TEST_ID'), eventData);
  expect(onPressMock).toHaveBeenCalledWith(eventData);
});
