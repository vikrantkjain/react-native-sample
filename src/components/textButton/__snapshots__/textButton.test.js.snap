// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextButton Render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="TEST_ID"
>
  <Text
    allowFontScaling={true}
    style={
      [
        undefined,
      ]
    }
  >
    hey
  </Text>
</View>
`;
